script.js

```diff
@@ -0,0 +1,245 @@
+const clearButton = document.querySelector("[data-clear]");
+const deleteButton = document.querySelector("[data-delete]");
+const plusminus = document.querySelector("[data-plusminus]");
+const inputNumberField = document.querySelector(".inputNumber");
+
+let inputOperator = undefined;
+let savedInputNumber = undefined;
+let currentNumber = 0;
+let resultNumber = undefined;
+
+let replaceNumber = true;
+
+document.body.onkeypress = (e) => {
+  if (isDigit(e.key)) {
+    checkInput(e.key);
+    keyPressEvent(e.key);
+  }
+};
+
+document.body.onkeydown = (e) => {
+  switch (e.key.toLowerCase()) {
+    case "enter":
+      calculate();
+      break;
+    case "backspace":
+      deleteNumber();
+      break;
+    case "c":
+      clearAll();
+      break;
+    case "_":
+      changeSign();
+      break;
+    case ".":
+      checkInput(".");
+      keyPressEvent(e.key);
+      break;
+    case "+":
+      setOperator("addition");
+      break;
+    case "-":
+      setOperator("subtraction");
+      break;
+    case "*":
+      setOperator("multiplication");
+      break;
+    case "/":
+      setOperator("division");
+      break;
+  }
+};
+
+deleteButton.onclick = () => {
+  deleteNumber();
+};
+
+clearButton.onclick = () => {
+  clearAll();
+};
+
+equalsButton.onclick = () => {
+  calculate();
+};
+
+plusminus.onclick = () => {
+  changeSign();
+};
+
+numberButtons.forEach((button) => {
+  button.onclick = () => {
+    checkInput(button.innerText);
+  };
+});
+
+operationButtons.forEach((button) => {
+  button.onclick = () => {
+    setOperator(button.dataset.operation);
+  };
+});
+
+const checkInput = (input) => {
+  if (input == ".") {
+    replaceNumber = false;
+    return checkDecimal(input);
+  }
+
+  if (inputNumberField.innerText == "0" && input == 0) {
+    replaceNumber = true;
+    return;
+  }
+
+  if (replaceNumber) {
+    replaceInput(input);
+    replaceNumber = false;
+  } else if (!replaceNumber) {
+    appendInput(input);
+  }
+
+  currentNumber = inputNumberField.innerText;
+};
+
+checkDecimal = (input) => {
+  if (inputNumberField.innerText.includes(".") && input == ".") {
+    return;
+  } else {
+    appendInput(input);
+  }
+};
+
+const calculate = () => {
+  switch (inputOperator) {
+    case "addition":
+      updateDisplay(calcAddition(savedInputNumber, currentNumber));
+      break;
+    case "subtraction":
+      updateDisplay(calcSubtraction(savedInputNumber, currentNumber));
+      break;
+    case "multiplication":
+      updateDisplay(calcMultiplication(savedInputNumber, currentNumber));
+      break;
+    case "division":
+      updateDisplay(calcDivision(savedInputNumber, currentNumber));
+      break;
+  }
+  resultNumber = inputNumberField.innerText;
+  savedInputNumber = resultNumber;
+  replaceNumber = true;
+  currentNumber = 0
+  removeActiveStyle();
+};
+
+const updateDisplay = (input) => {
+  if (input.toString().length >= 16) {
+    inputNumberField.innerText = parseFloat(input).toFixed(14);
```

Comment:
Maybe you could find a way to strip trailing zeros of a result? 7.2000000 should be displayed only as 7.2

Reply:
Fixed with the latest commit